sentry-samples/sentry-samples-android/src/main/java/io/sentry/samples/android/SecondActivity.kt

```diff
@@ -0,0 +1,112 @@
+package io.sentry.samples.android
+
+import android.content.Intent
+import android.os.Bundle
+import android.view.View
+import androidx.appcompat.app.AppCompatActivity
+import io.sentry.Sentry
+import io.sentry.SpanStatus
+import io.sentry.samples.android.databinding.ActivitySecondBinding
+import okhttp3.OkHttpClient
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+
+class SecondActivity : AppCompatActivity() {
+
+    private lateinit var repos: List<Repo>
+
+    private val client = OkHttpClient.Builder().addInterceptor(NetworkInterceptor()).build()
+
+    private lateinit var binding: ActivitySecondBinding
+
+    private val retrofit = Retrofit.Builder()
+            .baseUrl("https://api.github.com/")
+            .addConverterFactory(GsonConverterFactory.create())
+            .client(client)
+            .build()
+
+    private val service = retrofit.create(GitHubService::class.java)
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        val activeSpan = Sentry.getSpan()
+        val span = activeSpan?.startChild("onCreate", javaClass.simpleName)
+
+        binding = ActivitySecondBinding.inflate(layoutInflater)
+
+        binding.doRequest.setOnClickListener {
+            updateRepos()
+        }
+
+        binding.backMain.setOnClickListener {
+            finish()
+            startActivity(Intent(this, MainActivity::class.java))
+        }
+
+        // do some stuff
+
+        setContentView(binding.root)
+
+        span?.finish(SpanStatus.OK)
+    }
+
+    private fun showText(visible: Boolean = true, text: String = "") {
+        binding.text.text = if (visible) text else ""
+        binding.text.visibility = if (visible) View.VISIBLE else View.GONE
+
+        binding.indeterminateBar.visibility = if (visible) View.GONE else View.VISIBLE
+    }
+
+    override fun onResume() {
+        super.onResume()
+
+        val span = Sentry.getSpan()?.startChild("onResume", javaClass.simpleName)
+
+        // do some stuff
+
+        updateRepos()
+
+        // do some stuff
+
+        span?.finish(SpanStatus.OK)
+    }
+
+    private fun updateRepos() {
+        showText(false)
+
+        val currentSpan = Sentry.getSpan()
+        val span = currentSpan?.startChild("updateRepos", javaClass.simpleName)
+            ?: Sentry.startTransaction("updateRepos", "task")
+
+        service.listRepos(binding.editRepo.text.toString()).enqueue(object : Callback<List<Repo>> {
+            override fun onFailure(call: Call<List<Repo>>?, t: Throwable) {
+                span.finish(SpanStatus.INTERNAL_ERROR)
+                Sentry.captureException(t)
+
+                showText(true, "error: ${t.message}")
+
+                // be sure to finish all your spans before this
+                val transaction = Sentry.getSpan()
+                transaction?.finish(SpanStatus.INTERNAL_ERROR)
```

Comment:
Is this the place to finish the root span?
In case of the activity auto transaction wouldn't we be closing that prematurely?

There's time running from when we handle the HTTP failure and the user interaction like showing an error message so I believe we shouldn't close the transaction here.

Reply:
not sure if I follow your comment.

`updateRepos` is called during Activity.onResume which means the Activity is responsive and rendered.
if the API call returned a result or error, that's the end of my transaction, this is a sample app, I don't have any other screen to go nor a dialog to be shown.

the error message is updated via `showText` just before finishing the transaction.

@bruno-garcia can you elaborate?

Reply:
> the error message is updated via showText just before finishing the transaction.

Would this be the place to finish the transaction then?

I believe it would be at the very end of the transaction as in: Activity loaded, we queried the backend, got an error from it, checked the status code, prepared some error message and showed the dialog to the user: Finish.

But this is a sample and not a reason to block this PR so we can discuss this offline.